### ISSUE TITLE
Explain that when executing a rollback using `kubectl rollout undo` the undid version is excluded and cannot be applied anymore

### ISSUE THREAD

**Issue by xserrat** [kubernetes/website#26081]
**This is a Feature Request**

**What would you like to be added**

I would like to append extra info at the end of this section https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#rolling-back-to-a-previous-revision related to rollback to a previous revision, explaining that if you rollback to a previous revision using the following command
```bash
kubectl rollout undo deployment.v1.apps/nginx-deployment
```
the undid revision will be excluded and you cannot apply it again if it's needed.

**Why is this needed**

Because even if it's an understandable behaviour, it can cause confusion when:
* You list your revisions and see that the revision you've undone is not displayed.
* You try to apply the previously undid revision, the deployment does not apply the change.

**Comments**

This behaviour I found with my teammates reviewing our rollback process can be something useful to add in the documentation (if we're not wrong, of course) for other people.

If you think it's right, I can create a pull request to add this info in EN and ES languages.

**Comment by sftim** [comment]
Hi @xserrat 

This sounds like a useful explanation. Can you give an example of a command that you thought might work, but does not?

/language en
/kind feature

**Comment by xserrat** [comment]
Hi @sftim !

Yes, of course. While trying to reproduce the behaviour again with an example in Minikube, I found that the behaviour I mentioned above it's not exactly what it's happening:

Imagine we deploy a new release of our app, from version `v1.0` to `v1.1`. If we check our rollout history of the deployment we'll see something like this:

> kubectl rollout history deployment hello-world

```bash
deployment.apps/hello-world
REVISION  CHANGE-CAUSE
1         v1.0
2         v1.1
```

Then, if we want to roll back to the previous version `v.1.0` because we have a bug in the last version `v.1.1`:

> kubectl rollout undo deployment hello-world

```bash
hello-world
deployment.apps/hello-world rolled back
```

After that I check again the rollout history:

```bash
REVISION  CHANGE-CAUSE
2         v1.1
3         v1.0
```

As we see above, the revision 1 has been removed. It makes sense because the current revision (3) contains the same docker image as the revision 1.

This behaviour also occurs applying deployments, not only with rollbacks. Now, if we deploy the version `v1.1` again (editing the container image of the deployment resource):
> kubectl apply -f deployment.yaml

```bash
REVISION  CHANGE-CAUSE
3         v1.0
4         v1.1
```

In this case, due to the new version already exists (it was the revision 2), Kubernetes removes the revision 2 and creates the new one with the CHANGE-CAUSE of the revision 2 because they contain the same docker image and as a conclusion, it reuse the same deploy if it was deployed previously.

Anyway, I know that in the real world, we usually deploy incremental changes meaning that we always generate a different image tag so it's impossible reuse revisions/deployments. However, if you consider we can add some lines to make it more clear, let me know!
